cmake/qpoases_interface/qpoases_interface.cpp

```diff
@@ -30,26 +33,88 @@ This file is part of DQ Robotics.
 
 using namespace DQ_robotics;
 
-int main(void)
-{
+void test_solver(void) {
     DQ_QPOASESSolver qpoases_solver;
     DQ_SerialManipulatorDH robot = KukaLw4Robot::kinematics();
 
-    DQ_ClassicQPController classic_qp_controller(&robot,&qpoases_solver);
+    DQ_ClassicQPController classic_qp_controller(&robot, &qpoases_solver);
     classic_qp_controller.set_control_objective(ControlObjective::Pose);
     classic_qp_controller.set_gain(0.01);
     classic_qp_controller.set_damping(0.01);
 
     VectorXd theta_init(7);
-    theta_init << 0.,pi/4.,0.,0.,pi/4.,0.,0.;
+    theta_init << 0., pi / 4., 0., 0., pi / 4., 0., 0.;
 
     VectorXd theta_xd(7);
-    theta_xd << 0.,pi/2.,0.,0.,pi/2.,0.,0.;
+    theta_xd << 0., pi / 2., 0., 0., pi / 2., 0., 0.;
     DQ xd = robot.fkm(theta_xd);
 
-    VectorXd qd = classic_qp_controller.compute_setpoint_control_signal(theta_init,vec8(xd));
+    VectorXd qd = classic_qp_controller.compute_setpoint_control_signal(theta_init, vec8(xd));
 
     std::cout << "q_dot is " << qd.transpose() << std::endl;
+}
+
+
```

Comment:
@qlin960618 Please add the respective documentation for the function `are_vectors_equal()` and other functions you are adding. I'd recommend using lower-case naming for vectors to follow the style of the class `DQ_QPOASESSolver.h`. 

Example:

```cpp
/**
 * @brief // the documentation here
 * @param a // the documentation here
 * @param b // the documentation here
 * @param tolerance // the documentation here
 * @return  // the documentation here
 */
bool are_vectors_equal(const Eigen::VectorXd &a, const Eigen::VectorXd &b, double tolerance = 1e-3)
```